### ISSUE TITLE
co-pattern matching on empty record type removes body

### ISSUE THREAD

**Issue by ajrouvoet** [agda/agda#4536]
Understandably, co-pattern matching on a record type with no fields removes all equations:

```agda
record R : Set where

something : R
something = {! C-c C-c return here removes this line !} 
```

But Agda won't accept that declaration without a definition, preferring `something = record {}`

**Comment by jespercockx** [comment]
There already is an issue for this: https://github.com/agda/agda/issues/3006

**Comment by andreasabel** [comment]
This can be fixed without absurd copatterns (#3006), by producing
```agda
something = record {}
```
(as suggested in the OP).
